NiFiDeploy.groovy

```diff
@@ -568,6 +578,25 @@ nifiHostPort = nifiHostPort.endsWith('/') ? nifiHostPort[0..-2] : nifiHostPort
 assert nifiHostPort : "No NiFI REST API endpoint provided"
 
 nifi = new RESTClient("$nifiHostPort/nifi-api/")
+
+if (nifiHostPort.startsWith("https")) {
+  // add keystore
+  def keyStore = KeyStore.getInstance( KeyStore.defaultType )
+  keyStore.load(new FileInputStream(opts.keystore), opts.keypasswd.toCharArray())
+  // add trustStore
+  SSLSocketFactory sf;
+  if (opts.truststore) {
+    def trustStore = KeyStore.getInstance( KeyStore.defaultType )
+    trustStore.load(new FileInputStream(opts.truststore), opts.trustpasswd.toCharArray())
+     sf = new SSLSocketFactory(keyStore, opts.keypasswd, trustStore)
+  } else {
+    sf = new SSLSocketFactory(keyStore)
+  }
+  sf.setHostnameVerifier(SSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER)
```

Comment:
Agreed, though it most situations it seems a common problem. My original intention was to put this behind an option.